src/hooks/personalization/user-preferences.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from 'react';
+import { submitPersonalizationSelections } from '../../components/modal/personalization/utils';
+import { Tag } from '../../interfaces/tag';
+import { useNotificationContext } from '../../contexts/notification';
+
+interface UserPreferences {
+    followedTags: Array<Tag>;
+    emailPreference: boolean;
+}
+
+interface UserPreferencesHook {
+    updateUserPreferences: (prefs: UserPreferences) => Promise<void>;
+}
+const useUserPreferences = (): UserPreferencesHook => {
+    const { setNotification } = useNotificationContext();
+
+    const updateUserPreferences = useCallback(
+        async ({ followedTags, emailPreference }: UserPreferences) => {
+            try {
+                await submitPersonalizationSelections({
+                    followedTags,
+                    emailPreference,
+                });
+            } catch {
+                setNotification({
+                    message:
+                        'Your request could not be completed at this time. Please try again.',
+                    variant: 'WARN',
+                });
+                return;
+            }
+            setNotification({
+                message: 'Successfully saved your preferences',
+                variant: 'SUCCESS',
+            });
+        },
+        []
```

Comment:
is there a particular reason why `setNotification` isn't in the dependency array? if not, i think it's worth adding... not sure if react guarantees the context value is populated before every render, but i'm seeing a default `() => null` value for setNotification, so best to be safe imo

Reply:
Ahh yep I just missed that, will add that, thanks for the catch!